### ISSUE TITLE
About loss being NaN,  lr_scheduler.step(), optimizer.step()

### ISSUE THREAD

**Issue by FlyDre** [hkchengrex/STCN#116]
I've read issue #44. Like that case, I change the ResNet50 to another backbone.
So I check the link you mentioned:
https://discuss.pytorch.org/t/optimizer-step-before-lr-scheduler-step-error-using-gradscaler/92930/7

And therefore i change codes as below:
![change](https://user-images.githubusercontent.com/44801157/159656175-0302b549-f1fb-475c-82ea-be03f178fecd.png)

But seems losses( all 3 losses) still being NaN and the warning of "UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate" still exsists.

![problem](https://user-images.githubusercontent.com/44801157/159656394-c37c1ed0-3575-48fa-aee7-9b355399ef4c.png)

Is this normal or do I need to modify losses.py?

Thank you.

**Comment by hkchengrex** [comment]
The warning can be ignored. It doesn't matter.
I think the problem is in the backbone (I see that you are using ConvNext). I've also tried ConvNext, but
1) It gives NaN unless I turn off amp (--no_amp)
2) Even when it is working it is converging much slower than ResNet50
I would love to learn why/if you have a solution.

**Comment by FlyDre** [comment]
Thanks for replying.  I'd update this issue if I figure it out.
